File: apps/api/internal/platform/response/response.go
package response

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type Envelope struct {
	Data      any    `json:"data,omitempty"`
	Error     *Error `json:"error,omitempty"`
	RequestID string `json:"request_id,omitempty"`
}

type Error struct {
	Code    string `json:"code"`
	Message string `json:"message"`
}

func OK(c *gin.Context, data any) {
	c.JSON(http.StatusOK, Envelope{Data: data, RequestID: RequestID(c)})
}
func Created(c *gin.Context, data any) {
	c.JSON(http.StatusCreated, Envelope{Data: data, RequestID: RequestID(c)})
}
func Fail(c *gin.Context, status int, code, message string) {
	c.AbortWithStatusJSON(status, Envelope{Error: &Error{Code: code, Message: message}, RequestID: RequestID(c)})
}
func RequestID(c *gin.Context) string { return c.GetString("request_id") }

File: apps/ai/app/platform/response.py
"""JSON response envelope consistent with the Go backend.

Mirrors ``response.Envelope`` in the Go API: every response is
``{"data": ..., "error": {"code", "message"}, "request_id": ...}`` with the
unused field omitted.
"""

from __future__ import annotations

from typing import Any

from fastapi.responses import JSONResponse

from app.platform.logging import get_request_id


def _envelope(
    *,
    data: Any = None,
    error: dict[str, str] | None = None,
) -> dict[str, Any]:
    body: dict[str, Any] = {}
    if data is not None:
        body["data"] = data
    if error is not None:
        body["error"] = error
    request_id = get_request_id()
    if request_id:
        body["request_id"] = request_id
    return body


def ok(data: Any, status_code: int = 200) -> JSONResponse:
    return JSONResponse(status_code=status_code, content=_envelope(data=data))


def fail(status_code: int, code: str, message: str) -> JSONResponse:
    return JSONResponse(
        status_code=status_code,
        content=_envelope(error={"code": code, "message": message}),
    )
